tests/src/sanity.rs

```diff
@@ -257,7 +258,8 @@ mod tests {
     }
 
     async fn run(
```

Comment:
I think it's better to have `run_ls`/`run_exec` and then maybe share some code in external functions, but this current design is clumsy

Reply:
also I think it should be in a different test file as this file is getting too big already

Reply:
agree, will put it in a separate file